ceph_deploy/hosts/remotes.py

```diff
@@ -200,9 +200,15 @@ def write_monitor_keyring(keyring, monitor_keyring):
     write_file(keyring, monitor_keyring)
 
 
-def write_file(path, content):
+def write_file(path, content, mode=None, directory=None):
+    if directory:
+        if path.startswith("/"):
+            path = path[1:]
+        path = os.path.join(directory, path)
     with file(path, 'w') as f:
         f.write(content)
+    if mode:
+        os.chmod(path, mode)
```

Comment:
this isn't quite safe--an attacker could open the file after it is created but before it is chowned.  Even chowning before write isn't quite right.   You need to set the mode *at create time* to be totally safe.